### PR TITLE
Add Consul Service Mesh as an installable app

### LINKED ISSUE #202: [APP] Consul
Consul Service Mesh is a popular Service Mesh for Kubernetes, it would be awesome if Akade supported the installation.

## Expected Behaviour
`arkade install consul` would install and configure Consul Service Mesh for K8s

## Possible Solution
The Consul Helm chart [https://github.com/hashicorp/consul-helm](https://github.com/hashicorp/consul-helm) can be used to install consul. By default the Helm chart does not enable Consul Service Mesh, this can be enabled by an additional Helm values with sensible defaults.

Happy to PR this.

### PR DESCRIPTION
**Pull request by jsiebens**
## Description

Using the official HashiCorp Consul Helm chart, this PR adds Consul Service Mesh as an installable app.

By default a secured Consul Service Mesh is installed with:
- gossip encryption enabled
- TLS encryption enabled
- ACLs enabled
- Connect Injector enabled

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Fixes #202 

## How Has This Been Tested?

```
# start a kubernetes cluster
make build
./arkade install consul --help
./arkade install consul

kubectl get pods -n consul-system
kubectl port-forward -n consul-system consul-server-0 8501
# Open https://127.0.0.1:8501 in the browser
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [x] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by jsiebens** [comment]
@nicholasjackson what do you think about these default settings? 
Do you think of any other useful flags, such as enabling Mesh or Ingress Gateways?

**Review by Waterdrips — CHANGES_REQUESTED** [review]
Not looked too hard at the other code but we are trying to move over to the new framework for installing charts, https://github.com/alexellis/arkade/blob/master/pkg/apps/chart_app.go

There's an example in the loki_app that builds the `InstallerOptions` and passes them into that function. 

https://github.com/alexellis/arkade/blob/master/cmd/apps/loki_app.go#L71-L77 (example)

File: README.md [hunk 1]
```diff
@@ -248,6 +248,7 @@ Available Commands:
   argocd                  Install argocd
   cert-manager            Install cert-manager
   chart                   Install the specified helm chart
+  consul                  Install Consul
```
Comment (alexellis) [inline README.md:251]: Is this consul as in consul but on Kubernetes, or "consul-connect" as in the service mesh?

Comment (alexellis) [reply README.md:251]: @jsiebens PTAL

Comment (jsiebens) [reply README.md:251]: By default Consul Connect is enabled by the Helm chart. 
Will change it to reflect that...

**Comment by alexellis** [comment]
Commit merged into master via 55dfac7e82acd87e88c6c02066c54a31b9ac5ff8

**Pull request closed without merging**